database/api.go

```diff
@@ -0,0 +1,172 @@
+//go:generate mockgen -source=$GOFILE -package $GOPACKAGE -destination api.mock.gen.go
+package database
+
+import (
+	"github.com/beaconsoftwarellc/gadget/v2/database/qb"
+	"github.com/jmoiron/sqlx"
+)
+
+// Database is a reduced database interface which only exposes the transaction and select functions
+type API interface {
+	// Begin starts a transaction
+	Begin() error
+	// Commit commits the transaction
+	Commit() error
+	// Rollback aborts the transaction
+	Rollback() error
+
+	// Create initializes a Record and inserts it into the Database
+	Create(obj Record) error
```

Comment:
To satisfy the same interface as database, a lot of these methods are going to have to return a tracer error

Reply:
This will be needed to be able to use the database api in `AuthorizedRead`

Reply:
> To satisfy the same interface as database, a lot of these methods are going to have to return a tracer error

I think we want to avoid that, the interface does a good job of hiding tracer errors.


> This will be needed to be able to use the database api in `AuthorizedRead`

I'm happy to update `AuthorizedRead` ;)